### ISSUE TITLE
Clarification on Table 3 — Generalization from SceneFlow to DrivingStereo (EPE Metric)

### ISSUE THREAD

**Issue by Mdraqibkhan** [YaoChengTang/Diving-into-the-Fusion-of-Monocular-Priors-for-Generalized-Stereo-Matching#2]
Hi, thank you for sharing this impressive work! 

I have a question regarding Table 3:  Generalization from SceneFlow to DrivingStereo”, where EPE is reported as the evaluation metric.
I’m currently reproducing the results on the DrivingStereo dataset using models such as RAFT-Stereo, Selective-IGEV, etc., but I’m observing slight differences in the reported EPE values compared to those in the paper.

Could you please clarify or share: The exact evaluation protocol used on DrivingStereo (e.g., resolution, cropping, valid pixel mask, invalid disparity handling).
The formula or code used to compute EPE for this dataset.
Whether any scaling, disparity normalization, or postprocessing steps were applied before evaluation.
Having these details or a short evaluation script would be really helpful for reproducing your results accurately.

Thank you again for your excellent contribution and for any guidance you can provide! 

Best regards,
Raqib

**Comment by YaoChengTang** [comment]
Hi, thank you for your interest in our work!

As stated in Section 4.2 of our paper, both our method and all compared methods are evaluated without restrictions, such as limiting valid pixels to those with ground-truth disparity less than 192 or excluding occluded regions.

Below, we provide the updated evaluation code for the DrivingStereo dataset used in our experiments.

Notes:

- No scaling, disparity normalization, or postprocessing (e.g., filtering or resizing) is applied before evaluation.

- We use the half-resolution version (image_size='half').

If you find this helpful, please consider ⭐️ starring our repository. Thanks!



#### evaluate_stereo.py:

```python

@torch.no_grad()
def validate_drivingstereo(model, iters=32, root="", mixed_prec=False, subset='weather'):
    """ Peform validation using the drivingstereo dataset """
    model.eval()
    aug_params = {}
    val_dataset = datasets.DrivingStereo(aug_params, root=root, image_set=subset)
    torch.backends.cudnn.benchmark = True

    out_list, epe_list, elapsed_list = [], [], []
    for val_id in tqdm(range(len(val_dataset))):
        _, image1, image2, flow_gt, valid_gt, _ = val_dataset[val_id]
        image1 = image1[None].cuda()
        image2 = image2[None].cuda()
        padder = InputPadder(image1.shape, divis_by=32)
        image1, image2 = padder.pad(image1, image2)

        with autocast(enabled=mixed_prec):
            start = time.time()
            _, flow_pr = model(image1, image2, iters=iters, test_mode=True)
            end = time.time()

        if val_id > 50:
            elapsed_list.append(end-start)
        flow_pr = padder.unpad(flow_pr).cpu().squeeze(0)

        assert flow_pr.shape == flow_gt.shape, (flow_pr.shape, flow_gt.shape)
        epe = torch.sum((flow_pr - flow_gt)**2, dim=0).sqrt()

        epe_flattened = epe.flatten()
        val = valid_gt.flatten() >= 0.5

        out = (epe_flattened > 3.0)
        image_out = out[val].float().mean().item()
        image_epe = epe_flattened[val].mean().item()
        # if val_id < 9 or (val_id+1)%10 == 0:
            # print(f"DrivingStereo Iter {val_id+1} out of {len(val_dataset)}. EPE {round(image_epe,4)} D1 {round(image_out,4)}. Runtime: {format(end-start, '.3f')}s ({format(1/(end-start), '.2f')}-FPS)")
        epe_list.append(epe_flattened[val].mean().item())
        out_list.append(out[val].cpu().numpy())

    epe_list = np.array(epe_list)
    out_list = np.concatenate(out_list)

    epe = np.mean(epe_list)
    d1 = 100 * np.mean(out_list)

    avg_runtime = np.mean(elapsed_list)

    print(f"Validation DrivingStereo/{subset}: EPE {round(epe,4)}, D1 {round(d1,4)}, FPS {format(1/avg_runtime, '.2f')}, Time ({format(avg_runtime, '.3f')}s)")
    print("\r\n"*3)
    return {'DrivingStereo-epe': round(epe,4), 'DrivingStereo-d1': round(d1,4)}



if __name__ == '__main__':
    elif 'drivingstereo' in args.dataset:
    subset = args.dataset.split('_')[-1]
    # if args.root is None:
    if subset == 'weather':
        for s in ['foggy', 'rainy', 'cloudy', 'sunny']:
            print(f"######## Validating DrivingStereo {s} ########")
            res = validate_drivingstereo(model, iters=args.valid_iters, root=args.root, 
                                        mixed_prec=use_mixed_precision, subset=subset+'/'+s)
            print(f"##############################################\n")

```

#### stereo_dataset.py:

```python
class DrivingStereo(StereoDataset):
    def __init__(self, aug_params=None, root='datasets/DrivingStereo', image_set='training', image_size='half'):
        super(DrivingStereo, self).__init__(aug_params, sparse=True, reader=frame_utils.readDispDrivingStereoFull if image_size=='full' else frame_utils.readDispDrivingStereoHalf)
        root = root if len(root)>0 else DATASET_ROOT
        assert os.path.exists(root), "check the existence: {}".format(root)

        image1_list = sorted(glob(os.path.join(root, image_set, f'left-image-{image_size}-size/*')))
        image2_list = sorted(glob(os.path.join(root, image_set, f'right-image-{image_size}-size/*')))
        disp_list = sorted(glob(os.path.join(root, image_set, f'disparity-map-{image_size}-size/*')))
        print(f"image1_list: {len(image1_list)}")
        for idx, (img1, img2, disp) in enumerate(zip(image1_list, image2_list, disp_list)):
            self.image_list += [ [img1, img2] ]
            self.disparity_list += [ disp ]
```



#### frame_utils.py:

```python
def readDispDrivingStereoFull(filename):
    disp = cv2.imread(filename, cv2.IMREAD_ANYDEPTH) / 128.0
    valid = disp > 0.0
    return disp, valid

def readDispDrivingStereoHalf(filename):
    disp = cv2.imread(filename, cv2.IMREAD_ANYDEPTH) / 256.0
    valid = disp > 0.0
    return disp, valid
```

**Comment by YaoChengTang** [comment]
Since there's no further feedback, I'll go ahead and close this issue. Please feel free to reopen it if you have any other questions.
